relay-server/src/metric_stats.rs

```diff
@@ -109,6 +149,42 @@ impl MetricStats {
             metadata: Default::default(),
         })
     }
+
+    fn to_cardinality_metric(
+        &self,
+        limit: &CardinalityLimit,
+        report: &CardinalityReport,
+    ) -> Option<Bucket> {
+        let cardinality = report.cardinality;
+        if cardinality == 0 {
+            return None;
+        }
+
+        let name = report.name.as_ref()?;
+        let namespace = name.namespace();
+
+        if !namespace.has_metric_stats() {
+            return None;
+        }
+
+        let tags = BTreeMap::from([
+            ("mri".to_owned(), name.to_string()),
+            ("mri.namespace".to_owned(), namespace.to_string()),
+            (
+                "cardinality.window".to_owned(),
+                limit.window.window_seconds.to_string(),
+            ),
+        ]);
+
+        Some(Bucket {
+            timestamp: UnixTimestamp::now(),
+            width: 0,
+            name: cardinality_metric_mri(),
+            value: BucketValue::Gauge(GaugeValue::single(cardinality.into())),
```

Comment:
Why is this a gauge?

Reply:
Multiple Relay instances will report cardinality for the same window and tag combination. The only useful way to aggregate these multiple reports is by taking the `max`, for which a gauge is good enough.

Reply:
How is that different from a counter emitted every x seconds? I assume the gauges have different tag values for the pod names, so they don't aggregate together.

Reply:
The counter adds values, we specifically need the max per hour. So we might be emitting the values `1, 2, 3, 4, 5` and we're interested in just `5` not the sum of `15`.

Reply:
> need the max per hour

I assumed a lower granularity so the usage of a counter was maaaybe worth it